packages/health/example/lib/main.dart

```diff
@@ -92,63 +68,120 @@ class _HealthAppState extends State<HealthApp> {
     await Permission.activityRecognition.request();
     await Permission.location.request();
 
-    if (requested) {
-      try {
-        // fetch health data
-        List<HealthDataPoint> healthData = await health.getHealthDataFromTypes(yesterday, now, types);
+    // Check if we have permission
+    bool? hasPermissions = await health.hasPermissions(types, permissions: permissions);
 
-        // filter out duplicates
-        _healthDataList.removeWhere(
-            (shownElement) => healthData.indexWhere((newElement) => mapEquals(shownElement, newElement)) != -1);
+    // hasPermissions = false because the hasPermission cannot disclose if WRITE access exists.
+    // Hence, we have to request with WRITE as well.
+    hasPermissions = false;
```

Comment:
Why do we call 
bool? hasPermissions = await health.hasPermissions(types, permissions: permissions);
if one line later we set it to false?

Reply:
Yes, kinda stupid code from their plugin. Changed (But also, it is an example, no harm)